contracts/Vault.vy

```diff
@@ -785,6 +785,24 @@ def totalAssets() -> uint256:
     return self._totalAssets()
 
 
+@view
+@internal
+def _calculateLockedProfit() -> uint256:
+    lockedProfitCalculated: uint256 = 0
+    lockedFundsRatio: uint256 = (block.timestamp - self.lastReport) * self.lockedProfitDegration
+    precisionFactor: uint256 = self.precisionFactor
+    if(lockedFundsRatio < DEGREDATION_COEFFICIENT):
+        lockedProfitCalculated = self.lockedProfit - (
+                precisionFactor
+                * lockedFundsRatio
+                * self.lockedProfit
+                / DEGREDATION_COEFFICIENT
+                / precisionFactor
+            )
+            
+    return lockedProfitCalculated
```

Comment:
Just a little more efficient
```suggestion
    lockedFundsRatio: uint256 = (block.timestamp - self.lastReport) * self.lockedProfitDegration

    if(lockedFundsRatio < DEGREDATION_COEFFICIENT):
        precisionFactor: uint256 = self.precisionFactor
        return self.lockedProfit - (
                precisionFactor
                * lockedFundsRatio
                * self.lockedProfit
                / DEGREDATION_COEFFICIENT
                / precisionFactor
            )

    else:
        return 0
```

Reply:
Does it benefit to also save lockedProfit into memory instead of reading it twice?

`lockedProfit: uint256 = self.lockedProfit`

Reply:
yes it does! will add

Reply:
4165d877af771a7918d98630d7031705a74c1142